egs2/TEMPLATE/asr1/pyscripts/feats/dump_ssl_feature.py

```diff
@@ -0,0 +1,99 @@
+import argparse
+import logging
+import os
+
+import numpy as np
+from ssl_feature_utils import (
+    ESPnetHubertFeatureReader,
+    HubertFeatureReader,
+    MfccFeatureReader,
+    S3PRLFeatureReader,
+    dump_feature,
+    format_feature_conf_str,
+)
+
+from espnet2.utils.types import str2bool
+
+logging.basicConfig(
+    level=logging.DEBUG,
+    format="%(asctime)s (%(module)s:%(lineno)d) %(levelname)s: %(message)s",
+)
+logger = logging.getLogger("dump_hubert_or_wavlm_feature")
+
+
+feature_reader_choice = dict(
+    mfcc=MfccFeatureReader,
+    fairseq_hubert=HubertFeatureReader,
+    espnet_hubert=ESPnetHubertFeatureReader,
+    s3prl=S3PRLFeatureReader,
+)
+
+
+def get_parser():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--feature_conf", type=str, default=None)
+    parser.add_argument("--seed", default=0, type=int)
+    parser.add_argument("--use_gpu", default=True, type=str2bool)
+    parser.add_argument(
+        "--in_filetype",
+        type=str,
+        default="sound",
+        choices=["mat", "hdf5", "sound.hdf5", "sound"],
+        help="Specify the file format for the rspecifier. "
+        '"mat" is the matrix format in kaldi',
+    )
+    parser.add_argument(
+        "--out_filetype",
+        type=str,
+        default="npy",
+        choices=["npy", "mat", "hdf5"],
+        help="Specify the file format for the wspecifier. "
+        '"npy" is the matrix format in kaldi',
+    )
+    parser.add_argument(
+        "--write_num_frames", type=str, help="Specify wspecifer for utt2num_frames"
+    )
+    parser.add_argument(
+        "rspecifier", type=str, help="Read specifier for feats. e.g. ark:some.ark"
+    )
+    parser.add_argument(
+        "wspecifier", type=str, help="Write specifier. e.g. ark:some.ark"
+    )
+
+    return parser
+
+
+def main(args):
+    np.random.seed(args.seed)
```

Comment:
Which part of the SSL model is affected by the random seed?

Reply:
Good catch. I'll remove this one.